seqr/views/apis/saved_variant_api.py

```diff
@@ -0,0 +1,511 @@
+import logging
+import json
+from django.contrib.auth.decorators import login_required
+from django.db.models import Q
+from django.views.decorators.csrf import csrf_exempt
+
+from seqr.models import SavedVariant, VariantTagType, VariantTag, VariantNote, VariantFunctionalData, CAN_EDIT
+from seqr.model_utils import create_seqr_model, delete_seqr_model
+from seqr.utils.xpos_utils import get_chrom_pos
+from seqr.views.apis.auth_api import API_LOGIN_REQUIRED_URL
+from seqr.views.utils.json_to_orm_utils import update_model_from_json
+from seqr.views.utils.json_utils import create_json_response
+from seqr.views.utils.permissions_utils import get_project_and_check_permissions, check_permissions
+from xbrowse_server.base.models import Project as BaseProject, ProjectTag
+
+logger = logging.getLogger(__name__)
+
+
+@login_required(login_url=API_LOGIN_REQUIRED_URL)
+@csrf_exempt
+def saved_variant_data(request, project_guid):
+    project = get_project_and_check_permissions(project_guid, request.user)
+
+    variants = {}
+    variant_query = SavedVariant.objects.filter(project=project)\
+        .select_related('family')\
+        .only('genome_version', 'xpos_start', 'ref', 'alt', 'lifted_over_genome_version', 'lifted_over_xpos_start',
+              'saved_variant_json', 'family__guid', 'guid')\
+        .prefetch_related('varianttag_set', 'varianttag_set__created_by', 'varianttag_set__variant_tag_type',
+                          'variantfunctionaldata_set', 'variantfunctionaldata_set__created_by', 'variantnote_set',
+                          'variantnote_set__created_by')
+    if request.GET.get('family'):
+        variant_query = variant_query.filter(family__guid=request.GET.get('family'))
+    for saved_variant in variant_query:
+        variant_json = json.loads(saved_variant.saved_variant_json or '{}')
+        chrom, pos = get_chrom_pos(saved_variant.xpos)
+        genome_version = variant_json.get('extras', {}).get('genome_version', saved_variant.genome_version)
+        lifted_over_genome_version = saved_variant.lifted_over_genome_version or ('37' if genome_version == '38' else '38')
+        if saved_variant.lifted_over_xpos_start:
+            lifted_over_chrom, lifted_over_pos = get_chrom_pos(saved_variant.lifted_over_xpos_start)
+        else:
+            coords_field = 'grch%s_coords' % lifted_over_genome_version
+            coords = variant_json.get('extras', {}).get(coords_field, '').split('-')
+            lifted_over_chrom = coords[0].lstrip('chr')
+            lifted_over_pos = coords[1] if len(coords) > 1 else ''
+        variant = {
```

Comment:
I think it would actually be better to use an `orm_to_json_utils` function here because that camelcases the fields and also allows for some post prcessing